app/services/activitypub/process_status_update_service.rb

```diff
@@ -112,6 +112,8 @@ def update_media_attachments!
     @status.ordered_media_attachment_ids = @next_media_attachments.map(&:id)
 
     @media_attachments_changed = true if @status.ordered_media_attachment_ids != previous_media_attachments_ids
+
+    @status.media_attachments.reload if @media_attachments_changed
```

Comment:
This really is a nitpick, I think this is readable too but I would have written it this way:
```suggestion
    if @status.ordered_media_attachment_ids != previous_media_attachments_ids
      @media_attachments_changed = true
      @status.media_attachments.reload
    end
```